### ISSUE TITLE
Fix warnings inside our tests

### ISSUE THREAD

**Issue by danielduan** [storybooks/storybook#2196]
When running tests for cra-kitchen-sink, we have lots of warnings and messages thrown for react things.

Some of them are simply fixing the usage examples, some are resolving things react likes to complain about.

It'd be nice to have a clean test.

**Comment by syneva-runyan** [comment]
If no one's looking into this yet I can take a go at it

**Comment by danielduan** [comment]
@syneva-runyan if you can, that would be great. thanks!

**Comment by syneva-runyan** [comment]
@danielduan are the console.info statements good to be left in? For example, the ones that print out from app/react/src/server/babel_config.test.js

**Comment by danielduan** [comment]
I don't see any info statements in that file. Can you link it?

**Comment by dangreenisrael** [comment]
@danielduan @syneva-runyan  Any thoughts on making `console.error` and `console.warn` fail tests?

**Comment by Hypnosphi** [comment]
@dangreenisrael I think it's a good idea

**Comment by dangreenisrael** [comment]
@Hypnosphi Should this be onto master or release 3.3?

**Comment by danielduan** [comment]
I think tests that don't significantly impact the user's code can be put on master

**Comment by dangreenisrael** [comment]
https://github.com/storybooks/storybook/pull/2297

**Comment by syneva-runyan** [comment]
@danielduan, line 86 in app/react/src/server/babel_config.js for instance prints out => Loading custom .babelrc

`logger.info('=> Loading custom .babelrc');`

We could mock the logger so the messages don't show up in the console when running tests.

**Comment by danielduan** [comment]
that would be great, thanks!

**Comment by syneva-runyan** [comment]
Regarding a specific issue from tree/master/addons/info:

An Invalid DOM nesting error is being thrown in the first test," <pre> cannot be a decendant of <p>"

This error comes from the following:

```
withInfo(`
    '# Test story \n## with markdown info \ncontaing **bold**, *cursive* text and `code`'
`)
```

The provided markdown is rendered inside a <p> tag, as definied in text.js, <P>.

When the rest of the markdown parsed with marksy, the `code` portion of the info is rendered in a <pre> tag, which causes the invalid DOM nesting error.

A similar error would be thrown if a link was provided in the markdown, ex
```
withInfo(`
    '# Test story \n## with markdown info \ncontaing [[I'm an inline-style link with title](https://www.google.com "Google's Homepage")]'
`)
```

Possible solutions to this problem would be:

+ Change <P> to render content inside of a <div> or other containing element.
+ Change <Code> to render content inside a <span> or other element. (Would not fix the link scenario)
+ A technical solution involving understanding what markdown is being passed to component and responding appropriately or throwing an error.

Thoughts on the best approach here? @danielduan

**Comment by danielduan** [comment]
It should probably just be a `<div>` if there aren't any huge regressions visually.

**Comment by syneva-runyan** [comment]
Raised a pull request, https://github.com/storybooks/storybook/pull/2343 Sorry for taking so long!  

Regarding warning from `addons/info/scr/index.test.js` line 10:

I thought about putting in a fix for an issue that came up here where using marksy to output a 'function' spat out a warning, but figured that may be a separate issue (or maybe a marksy issue?) to deal with.  Instead, I've added a .toString() to the passed function, suppressing the warning for now.

**Comment by dangreenisrael** [comment]
If this is merged into master, should the issue be closed? @Hypnosphi. This (https://github.com/storybooks/storybook/issues/1138) seems to imply that we want to get issues closed on the sooner side.

**Comment by Hypnosphi** [comment]
Let's close it when the same is done for `release/3.3`

**Comment by syneva-runyan** [comment]
Put in pull request for the warnings in release/3.3 https://github.com/storybooks/storybook/pull/2381/files

**Comment by Hypnosphi** [comment]
Fixed with #2343 and #2381
